extensions/amp-base-carousel/1.0/scroller.js

```diff
@@ -35,13 +35,7 @@ const RESET_SCROLL_REFERENCE_POINT_WAIT_MS = 200;
  * @param {!BaseCarouselDef.ScrollerProps} props
  * @return {PreactDef.Renderable}
  */
-export function Scroller({
-  children,
-  loop,
-  restingIndex,
-  setRestingIndex,
-  scrollRef,
-}) {
+export function Scroller({children, loop, ref, restingIndex, setRestingIndex}) {
```

Comment:
Hmm. Naming and structure-wise, the pattern actually goes a bit different. The most common think I saw:

```
function ScrollerWithRef({...}, ref) {
}

export const Scroller = forwardRef(ScrollerWithRef);
```

The key things here:
1. The module only exports the result of the `forwardRef`. It's somewhat important to "hide" the `ScrollerWithRef` so that no one case use it directly.
2. The `displayName` in this case shouldn't be needed since the name will be `Scroller` due to the declaration.

See this for more details: https://reactjs.org/docs/hooks-reference.html#useimperativehandle

Reply:
1. We have an eslint rule that doesn't let us export the result of `forwardRef`: `Cannot export side-effect eslint(local/no-export-side-effect)` One option is to move both `Scroller` and `ScrollerWithRef` to `base-carousel.js`, but as it stands the current separation across files is to avoid this rule. 
2. The `displayName` is `ForwardRef` when using a `forwardRef`, so that is why it is set.

Reply:
IMHO we need to whitelist `forwardRef` as an allowed artifact. @jridgewell how can we do this?